github_connector/models/abstract_github_model.py

```diff
@@ -0,0 +1,278 @@
+        to mention which items should be synchronized from github when the
+        user click on 'Full Update' Button"""
+        pass
+
+    @api.multi
+    def _hook_after_github_creation(self):
+        """Hook that will be called, after a creation in github.
+        Override this function to add custom logic for after creation."""
+        pass
+
+    # Custom Public Function
+    @api.model
+    def get_from_id_or_create(self, data, extra_data=None):
+        """Search if the odoo object exists in database. If yes, returns the
+            object. Otherwise, creates the new object.
+
+            :param data: dict with github 'id' and 'url' keys
+            :return: The searched or created object
+
+            :Example:
+
+            >>> self.env['github_organization'].get_from_id_or_create(
+                {'id': 7600578, 'url': 'https://api.github.com/orgs/OCA'})
+        """
+        extra_data = extra_data and extra_data or {}
+
+        # We try to search object by id
+        existing_object = self.search(
+            [('github_id_external', '=', data['id'])])
+        if existing_object:
+            return existing_object
+
+        # We try to see if object exist by name (instead of id)
+        if self._github_login_field and self._github_login_field in data:
+            existing_object = self.search([
+                ('github_login', '=', data[self._github_login_field])])
+            if existing_object:
+                # Update the existing object with the id
+                existing_object.github_id_external = data['id']
+                _logger.info(
+                    "Existing object %s#%d with github name '%s' has been"
+                    " updated with unique github id %s#%s" % (
+                        self._name, existing_object.id,
+                        data[self._github_login_field], data['id'],
+                        self.github_type()))
+                return existing_object
+
+        if self._need_individual_call:
+            github_connector = self.get_github_connector(self.github_type())
+            data = github_connector.get_by_url(data['url'], 'get')
+        return self._create_from_github_data(data, extra_data)
+
+    @api.model
+    def create_from_name(self, name):
+        """Call Github API, using a URL using github name. Load data and
+            Create Odoo object accordingly, if the odoo object doesn't exist.
+
+            :param name: the github name to load
+            :return: The created object
+
+            :Example:
+
+            >>> self.env['github_organization'].create_from_name('OCA')
+            >>> self.env['github_repository'].create_from_name('OCA/web')
+        """
+        github_connector = self.get_github_connector(self.github_type())
+        res = github_connector.get([name])
+        # search if ID doesn't exist in database
+        current_object = self.search([('github_id_external', '=', res['id'])])
+        if not current_object:
```

Comment:
If current_object is not set i get this:

```
2017-10-26 07:30:34,908 9324 ERROR github1 odoo.http: Exception during JSON request handling.
Traceback (most recent call last):
  File "/home/onestein/Pycharm/Gitlab/odoo-10.0/src/odoo-10.0/odoo/http.py", line 638, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/home/onestein/Pycharm/Gitlab/odoo-10.0/src/odoo-10.0/odoo/http.py", line 675, in dispatch
    result = self._call_function(**self.params)
  File "/home/onestein/Pycharm/Gitlab/odoo-10.0/src/odoo-10.0/odoo/http.py", line 331, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/home/onestein/Pycharm/Gitlab/odoo-10.0/src/odoo-10.0/odoo/service/model.py", line 119, in wrapper
    return f(dbname, *args, **kwargs)
  File "/home/onestein/Pycharm/Gitlab/odoo-10.0/src/odoo-10.0/odoo/http.py", line 324, in checked_call
    result = self.endpoint(*a, **kw)
  File "/home/onestein/Pycharm/Gitlab/odoo-10.0/src/odoo-10.0/odoo/http.py", line 933, in __call__
    return self.method(*args, **kw)
  File "/home/onestein/Pycharm/Gitlab/odoo-10.0/src/odoo-10.0/odoo/http.py", line 504, in response_wrap
    response = f(*args, **kw)
  File "/home/onestein/Pycharm/Gitlab/odoo-10.0/src/odoo-10.0/addons/web/controllers/main.py", line 889, in call_button
    action = self._call_kw(model, method, args, {})
  File "/home/onestein/Pycharm/Gitlab/odoo-10.0/src/odoo-10.0/addons/web/controllers/main.py", line 877, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "/home/onestein/Pycharm/Gitlab/odoo-10.0/src/odoo-10.0/odoo/api.py", line 681, in call_kw
    return call_kw_multi(method, model, args, kwargs)
  File "/home/onestein/Pycharm/Gitlab/odoo-10.0/src/odoo-10.0/odoo/api.py", line 672, in call_kw_multi
    result = method(recs, *args, **kwargs)
  File "/home/onestein/Pycharm/Gitlab/odoo-10.0/src/oca/github_connector/models/wizard_load_github_model.py", line 35, in button_create_from_github
    my_obj.update_from_github(True)
AttributeError: 'NoneType' object has no attribute 'update_from_github'
```



I think you miss this line at the end of the method:

```
return current_object
```